src/ssd/ssd-extents.c

```diff
@@ -111,14 +76,15 @@ ssd_extents_update(struct ssd *ssd)
 	int full_width = width + 2 * theme->border_width;
 	int extended_area = SSD_EXTENDED_AREA;
 	int corner_size = extended_area + theme->border_width +
-		theme->window_button_width / 2;
+		MIN(theme->window_button_width, width) / 2;
```

Comment:
This fixes the overlap of corner extents and negative widths of top/bottom extents.